mbed_lstools/lstools_base.py

```diff
@@ -73,6 +73,8 @@ def __init__(self):
         self.lock_file = os.path.join(MbedLsToolsBase.HOME_DIR,
                                       MbedLsToolsBase.MBEDLS_HOME_DIR, MbedLsToolsBase.MBEDLS_GLOBAL_LOCK)
         self.mbedls_get_mocks()
+        # Make sure retargeting is applied if retarget file exists
+        self.retarget(True)
```

Comment:
@mazimkhan Making 'opts' globally available doesn't make a whole lot of sense, since mbedls isn't guaranteed to run commandline. Other mbed tools just call the create() method directly to get the mbeds object and run functions on that. No 'opts' are passed.

I'd be perfectly happy if we deprecate the skip_retarget option and just always apply retargeting. As said in my previous comment, if a user wants to stop retargeting, he/she should just remove/rename the retarget file or the relevant entry inside of it...